docs/guides/expectations/creating_custom_expectations/how_to_create_custom_column_pair_map_expectations.md

```diff
@@ -0,0 +1,324 @@
+import TechnicalTag from '@site/docs/term_tags/_tag.mdx';
+
+**`ColumnPairMapExpectations`** are one of the most common types of <TechnicalTag tag="expectation" text="Expectation" />. They are evaluated for a pair of columns and ask a yes/no question about the row-wise relationship between those two columns. Based on the result, they then calculate the percentage of rows that gave a positive answer. If the percentage is high enough, the Expectation considers that data valid.
+
+This guide will walk you through the process of creating a custom `ColumnPairMapExpectation`.
+
+<Prerequisites>
+
+- Read the [overview for creating Custom Expectations](./overview.md).
+
+</Prerequisites>
+
+## Steps
+
+### 1. Choose a name for your Expectation
+
+First, decide on a name for your own Expectation. By convention, `ColumnPairMapExpectations` always start with `expect_column_pair_values_`. You can see other naming conventions in the [Expectations section](../../../contributing/style_guides/code_style.md#expectations) of the code Style Guide.
+
+Your Expectation will have two versions of the same name: a `CamelCaseName` and a `snake_case_name`. For example, this tutorial will use:
+
+- `ExpectColumnPairValuesToHaveADifferenceOfThree`
+- `expect_column_pair_values_to_have_a_difference_of_three`
+
+### 2. Copy and rename the template file
+
+By convention, each Expectation is kept in its own python file, named with the snake_case version of the Expectation's name.
+
+You can find the template file for a custom [`ColumnPairMapExpectation` here](https://github.com/great-expectations/great_expectations/blob/develop/examples/expectations/column_pair_map_expectation_template.py). Download the file, place it in the appropriate directory, and rename it to the appropriate name.
+
+```bash
+cp column_pair_map_expectation_template.py /SOME_DIRECTORY/expect_column_pair_values_to_have_a_difference_of_three.py
+```
+
+<details>
+  <summary>Where should I put my Expectation file?</summary>
+  <div>
+    <p>
+        During development, you don't actually need to put the file anywhere in particular. It's self-contained, and can be executed anywhere as long as <code>great_expectations</code> is installed.
+    </p>
+    <p>
+        But to use your new Expectation alongside the other components of Great Expectations, you'll need to make sure the file is in the right place. The right place depends on what you intend to use it for.
+    </p>
+    <p>
+        <ul>
+            <li>If you're building a <TechnicalTag tag="custom_expectation" text="Custom Expectation" /> for personal use, you'll need to put it in the <code>great_expectations/plugins/expectations</code> folder of your Great Expectations deployment, and import your Custom Expectation from that directory whenever it will be used. When you instantiate the corresponding <code>DataContext</code>, it will automatically make all <TechnicalTag tag="plugin" text="Plugins" /> in the directory available for use.</li>
```

Comment:
I would remove the two preceding paragraphs and just start this section with line 49.

Reply:
I think I would merge the two preceding paragraphs, but keep the basics of them.  We don't want to encourage practices that could lead to confusion (such as someone trying to use GE for production work from a copy of the development repo because they put their Expectation in a strange place to start with) so I think that the content of the first paragraph should specify the difference between developing a plugin in the dev repo and using the plugin within a production installation of GE a little more clearly, too.

That said, the "The right place depends on what you intend to use it for" line sets up the user to expect a series of use cases; the jump between "If you're building a Custom Expectation for personal use" to "If you're building a Custom Expectation to contribute to an open source project" is a little more jarring without it.

I would use something along the lines of:

Within a development environment, you don't need to put the file in a specific folder.  The Expectation itself should be self-contained, and can be executed anywhere as long as <code>great_expectations</code> is installed, which is sufficient for development and testing.  However, to use your new Expectation alongside the other components of Great Expectations (as one would for production purposes), you'll need to make sure the file is in the right place.  Where the right place is will depend on what you intend to use it for.